java/src/org/openqa/selenium/bidi/browsingcontext/BrowsingContext.java

```diff
@@ -0,0 +1,238 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.bidi.browsingcontext;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WindowType;
+import org.openqa.selenium.bidi.BiDi;
+import org.openqa.selenium.bidi.Command;
+import org.openqa.selenium.bidi.HasBiDi;
+import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.json.Json;
+import org.openqa.selenium.json.JsonInput;
+import org.openqa.selenium.json.TypeToken;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class BrowsingContext {
+
+  private final String id;
+  private final BiDi bidi;
+
+  private static final String CONTEXT = "context";
+  private static final String RELOAD = "browsingContext.reload";
+  private static final String HANDLE_USER_PROMPT = "browsingContext.handleUserPrompt";
+
+  protected static final Type LIST_OF_BROWSING_CONTEXT_INFO =
+    new TypeToken<List<BrowsingContextInfo>>() {
+    }.getType();
+
+  private final Function<JsonInput, String> browsingContextIdMapper = jsonInput -> {
+    Map<String, Object> result = jsonInput.read(Map.class);
+    return (String) result.get(CONTEXT);
```

Comment:
Maybe we could call the `toString()` instead of doing a cast ? The `toString()` would avoid `ClassCastException`